### ISSUE TITLE
[Bug][zos_copy] When `src` is a directory and `dest` is a directory, if mode is set they are ignored

### ISSUE THREAD

**Issue by ddimatos** [ansible-collections/ibm_zos_core#411]
##### SUMMARY
When copying a directory from the controller to an0ther directory  (same name on the target) with mode set, the directory ends up being created with what ever the umask is on the target and not the mode specified. 

##### COMPONENT NAME
zos_copy
##### ANSIBLE VERSION
2.11.2 and 2.9.27

##### SPECIFY ANSIBLE COLLECTION VERSION
v1.3.5

##### SPECIFY THE Z OPEN AUTOMATION UTILITIES VERSION
1.2.0

##### STEPS TO REPRODUCE

Playbook recreates the error, there are two tasks with a `when: 0 >1` that should be enabled only when understood what they do. 
```yaml

---
- hosts: zvm
  collections:
    - ibm.ibm_zos_core
  gather_facts: no
  environment: "{{ environment_vars }}"

  tasks:
    # Enable this by changing the when clause , otherwise this intentionally is skipped
    - name: Delete the parent `/u/dir2test` directory if it exists using ansible builtin file module
      ansible.builtin.file:
        path: /u/dir2test
        state: absent
      ignore_errors: yes
      when: 0 > 1

    - name: Create the directory if it does not exist and set file permissions to 775
      ansible.builtin.file:
        path: "{{ item }}"
        state: directory
        mode: '0775'
      register: file_output
      loop:
        - /u/dir2test/products/
        - /u/dir2test/products/zosmf
        - /u/dir2test/products/zosmf/workflows
    - name: Result for create directory task
      debug:
        var: file_output

    - name: Display directory stat(s) for one or more directories
      ansible.builtin.stat:
        path: "{{ item }}"
      register: file_stat
      loop:
        - /u/dir2test/products/
        - /u/dir2test/products/zosmf
        - /u/dir2test/products/zosmf/workflows
    - name: Result for directory stat(s)
      debug:
        var: file_stat

    - name: Create a directory on the controller for the recreation of this issue
      ansible.builtin.file:
        path: /tmp/workflows
        state: directory
      delegate_to: localhost

    - name: Create a series of files in the controller under /tmp/workflows
      file:
        path: "/tmp/workflows/flow-{{ item }}.xml"
        state: touch
      with_sequence: count=4
      register: result
      delegate_to: localhost

    - name: Copy workflows directory as a 644 to directory zosmf
      zos_copy:
        src: /tmp/workflows
        dest: /u/dir2test/products/zosmf/workflows
        mode: '0644'
        group: 'USS1GRP'
        owner: 'USS1ADM'
        validate: true
        ignore_sftp_stderr: true
        force: true
      register: file_output

    - name: Results for file copies
      debug:
        var: file_output


    ############################################################################
    # This is the zos_copy work around for now, consider option `recurse` if you
    # need nested directories to change permission, see doc:
    # https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-recurse
    ############################################################################
    - name: "[PERMISSION OVERRIDE] Change the directory permissions after the zos_copy module had changed them."
      ansible.builtin.file:
        path: /u/dir2test/products/zosmf/workflows
        state: directory
        mode: '0775'
      when: 0 > 1

    ############################################################################
    # I expect /u/dir2test/products/zosmf to stat fine with a 755 as zos_copy
    ############################################################################
    - name: Display directory stat for one or more directories
      ansible.builtin.stat:
        path: "{{ item }}"
      register: file_stat
      loop:
        - /u/dir2test/products
        - /u/dir2test/products/zosmf
    - name: Result for directory stat(s)
      debug:
        var: file_stat

    ############################################################################
    # I expect this to fail because zos_copy was asked to copy the workflows
    # directory as a 644, no stat will succeed here. This is
    # as designed, https://linuxize.com/post/understanding-linux-file-permissions/
    #
    # Although this does not fail because zos_copy is not honoring the mode
    # permissions supplied for directories, it does fine for files but the code
    # is not honoring directories.
    ############################################################################
    - name: Display directory stat for one or more directories
      ansible.builtin.stat:
        path: /u/dir2test/products/zosmf/workflows
      register: file_stat

    - name: Result for directory stat(s)
      debug:
        var: file_stat

    - name: Response for verifying actionable messages on system {{ system_name }} are canceled
      assert:
        that:
          - file_stat is defined and  (file_stat.stat.mode == "0775")
        success_msg: "The directory /u/dir2test/products/zosmf/workflows has correct permissions {{ file_stat.stat.mode }}"
        fail_msg: "The directory /u/dir2test/products/zosmf/workflows has INCORRECT permissions {{ file_stat.stat.mode }} "
      ignore_errors: yes
```

##### EXPECTED RESULTS
I expect mode to be honored for directories, only honored for files.

**Comment by ddimatos** [comment]
Should be fixed in upcoming 1.3.6 and 1.4.0-beta.2

**Comment by ddimatos** [comment]
Checked into 1.3.6 and 1.4.-beta.2 branches.
